src/HttpBodyStream.php

```diff
@@ -0,0 +1,254 @@
+<?php
+namespace Legionth\React\Http;
+
+use Psr\Http\Message\StreamInterface;
+use React\Stream\ReadableStreamInterface;
+use React\Stream\WritableStreamInterface;
+use Evenement\EventEmitter;
+use React\Stream\Util;
+
+class HttpBodyStream extends EventEmitter implements StreamInterface, ReadableStreamInterface
+{
+    private $input;
+    private $closed;
+    private $closing = false;
+
+    public function __construct(ReadableStreamInterface $input)
+    {
+        $this->input = $input;
+
+        $this->input->on('data', array($this, 'handleData'));
+        $this->input->on('end', array($this, 'handleEnd'));
+        $this->input->on('error', array($this, 'handleError'));
+        $this->input->on('close', array($this, 'close'));
+    }
+
+    public function handleData($data)
```

Comment:
Yes these ar public because, I use them like this `$this->input->on('data', array($this, 'handleData'));
` in the constructor. I could define them as an anonymous function, in the constructor, if you think this would be better.